File: backend/app/models/driver.py
from pydantic import BaseModel


class Driver(BaseModel):
    driver_number: int
    full_name: str
    name_acronym: str
    team_name: str
    team_colour: str
    first_name: str | None = None
    last_name: str | None = None
    headshot_url: str | None = None
    country_code: str | None = None


class DriversResponse(BaseModel):
    drivers: list[Driver]


class HeadToHeadDriver(BaseModel):
    name_acronym: str
    full_name: str
    qualifying_wins: int
    race_wins: int
    avg_position: float
    total_points: float


class HeadToHeadResponse(BaseModel):
    year: int
    team_name: str
    team_colour: str
    driver_1: HeadToHeadDriver
    driver_2: HeadToHeadDriver

File: frontend/src/types/driver.ts
export interface Driver {
  driver_number: number;
  full_name: string;
  name_acronym: string;
  team_name: string;
  team_colour: string;
  first_name: string;
  last_name: string;
  headshot_url: string | null;
  country_code: string | null;
}

export interface HeadToHead {
  year: number;
  team_name: string;
  team_colour: string;
  driver_1: HeadToHeadDriver;
  driver_2: HeadToHeadDriver;
}

export interface HeadToHeadDriver {
  name_acronym: string;
  full_name: string;
  qualifying_wins: number;
  race_wins: number;
  avg_position: number;
  total_points: number;
}
